lib/media/buffering_observer.js

```diff
@@ -67,9 +67,14 @@ shaka.media.BufferingObserver = class {
     const threshold = this.thresholds_.get(this.previousState_);
 
     const oldState = this.previousState_;
-    const newState = (bufferedToEnd || bufferLead >= threshold) ?
+    let newState = (bufferedToEnd || bufferLead >= threshold) ?
```

Comment:
I think this would be clearer if, instead of having a separate clause for `threshhold == 0` with subtly different logic, you instead just do:

```js
const newState = (bufferedToEnd || (bufferLead >= threshold && bufferLead > 0)) ?
                 (State.SATISFIED) :
                 (State.STARVING);
```

Reply:
Done!